### DISCUSSION TITLE
extract participating collaborator dataset size in a federation round in the aggregator function

### DISCUSSION THREAD

**Discussion by dskhanirfan** [securefederatedai/openfederatedlearning#336] [General]
Hi
How can I extract each participating collaborator dataset size in a federation round in the aggregator function? Is ShardDescriptor and ShardDataset available in aggregator function? it available in db_iterator?

**Comment by itrushkin** [comment]
Hi @dskhanirfan! `ShardDescriptor` and `ShardDataset` are not available in `AggregationFunction`, but every `LocalTensor` that participates in aggregation has its weight equal to **relative** data size. For example, if federation consists of 100 samples divided equally  across 10 collaborators, `weight`s of all `local_tensors` in aggregation function would be equal to 0.1. And default weighted averaging (`WeightedAverage` in OpenFL) sums these local tensors with corresponding relative weights. Do you really need **absolute** values for tensor weights?

**Reply by dskhanirfan** [reply]
Hi @itrushkin  I require the absolute dataset size of all participating collaborators in the aggregator function
